### PR TITLE
[tests] update OTBR CI test script

### PR DESCRIPTION
**Pull request by wgtdkp**
This PR updates the OTBR CI test script.

Depends-On: openthread/ot-br-posix#712

### CODE REVIEW

File: script/test [hunk 1]
```diff
@@ -262,17 +262,9 @@ do_build_otbr_docker()
     echo "Building OTBR Docker ..."
     local otdir
     local otbrdir
-    local otbr_options="-DOT_DNSSD_SERVER=ON -DOT_DNS_CLIENT=ON -DOT_SRP_CLIENT=ON -DOT_SLAAC=ON -DOT_DUA=ON -DOT_MLR=ON -DOT_COVERAGE=ON -DOTBR_REST=OFF -DOTBR_WEB=OFF -DOTBR_MDNS=mDNSResponder -DOTBR_SRP_ADVERTISING_PROXY=ON"
+    local otbr_options="-DOT_DNSSD_SERVER=ON -DOT_DNS_CLIENT=ON -DOT_SRP_CLIENT=ON -DOT_SLAAC=ON -DOT_DUA=ON -DOT_MLR=ON -DOT_COVERAGE=ON -DOTBR_REST=OFF -DOTBR_WEB=OFF"
```
Comment (simonlingoogle) [inline script/test:265]: nit: can we use a array for otbr_options? It's getting too long.

Comment (simonlingoogle) [inline script/test:265]: Should we still use `-DOTBR_MDNS=mDNSResponder` ? Otherwise how do we specify the mDNS implementation?

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1: 

with nits

File: script/test (hunk 1 above)
Comment (wgtdkp) [reply script/test:265]: I tried it but got errors with the docker command. I was not sure about the cause but just kept it as is :)

Comment (wgtdkp) [reply script/test:265]: This is handle the the OTBR setup script. This is also the purpose of the PR: simplify the OTBR docker build and ensures the OTBR changes is working.

**Review by jwhui — APPROVED** [review]

**Pull request merged**